### DISCUSSION TITLE
LoRa Integration

### DISCUSSION THREAD

**Discussion by cpoore1** [ainfosec/FISSURE#28] [General]
Also refer to the LoRa Discord channel.

LoRa integration into FISSURE is of interest to people. There's nothing in there that addresses LoRa so I'm starting fresh. Does anyone have hardware recommendations for generating signals that I can quickly test against? Has anyone had a positive experience with any of these software programs:
- https://github.com/jieter/python-lora
- https://github.com/anthonykirby/lora-packet
- https://github.com/JiapengLi/lorawan-parser
- https://github.com/anthonykirby/lora-packet/blob/master/README.md
- https://github.com/brocaar/lorawan
- https://github.com/rpp0/gr-lora
- https://github.com/tapparelj/gr-lora_sdr
- https://github.com/PentHertz/LoRa_Craft
- https://github.com/nootropicdesign/lora-mesh
- https://github.com/BastilleResearch/gr-lora
- https://github.com/myriadrf/LoRa-SDR
- https://www.thethingsindustries.com/docs/reference/data-formats/
- https://github.com/rpp0/gr-lora-samples

**Comment by DanielRIOT** [comment]
LoRa is the "low level" modulation ( chirp spread spectrum ) aspect ( point to point RF )
LoRaWAN is the "network" aspect ( addressing, routing, encryption, gateways, "SSID"s  )

* https://github.com/brocaar/lorawan : is the Server infrastructure for a LoRaWAN network ( authentication, database, encryption, etc - like the "RADIUS" and  "DHCP", Firewall equivalent in WLAN networks ) 

Gateways and endnodes transmit data differently ( to distinguish "upstream" ie. sensor to server via  gateway and "downstream" ie. server to sensor via gateway traffic ) i think I and Q are swopped 

in a LoRaWAN network sensors do not communicate directly with each other, everything goes via the network server and gateways ( LoRa rf to IP converters )

if you use "just LoRa " then direct comms is possible ( like using https://www.airspayce.com/mikem/arduino/RadioHead/ library for *any* other rf module ), but this mode is incompatible with LoRaWAN

This is also a good article from gnuradio 
* https://pubs.gnuradio.org/index.php/grcon/article/download/8/7/

for LoRaWAN packets, you have a few easy-ish options and buying minimal hardware:
using a LoraFeather or pycom LoRa sensor/ endnode *thing* 
* a free account on TTN if there is coverage in your area : https://www.thethingsnetwork.org/
* a personal LoRaWAN network using a single gateway with built in server : https://www.multitech.com/brands/multiconnect-conduit
*  a personal LoRaWAN network using brocaar server and *any* gateway that can speak to it ( like RAK, multitech, Rpi with a usb gateway card )

Lats Time i checked ( more than a year ago ) gr_lora could not decode multiple LoraWAN packets "efficiently" if they were sent at different chirp rates unless you had a fairly beefy machine and ran 8 or so decoders in parralel, 

LoraWAN also has band plans for each region ( which channels are uplink and registration channels, and which ones are "hopped to" when sending data

I have not looked at LoRa in over a year so things might have changed since the V1.1 spec
